cdec/installation-manager-cli-bundle/src/main/java/com/codenvy/im/cli/preferences/CodenvyOnpremPreferences.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.im.cli.preferences;
+
+import com.codenvy.cli.command.builtin.MultiRemoteCodenvy;
+import com.codenvy.cli.command.builtin.Remote;
+import com.codenvy.cli.preferences.Preferences;
+import com.codenvy.cli.security.RemoteCredentials;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * @author Dmytro Nochevnov
+ */
+public class CodenvyOnpremPreferences {
+    public static final String CODENVY_ONPREM_REMOTE_NAME = "codenvy-onprem";
+
+    private final Preferences globalPreferences;
+    private final String remote = CODENVY_ONPREM_REMOTE_NAME;
+
+    private       MultiRemoteCodenvy storage;
+
+    public CodenvyOnpremPreferences(Preferences globalPreferences, MultiRemoteCodenvy storage) {
+        this.globalPreferences = globalPreferences;
+        this.storage = storage;
+    }
+
+    /**
+     * Ensure Codenvy on-prem preferences has url = codenvyOnpremUrl.
+     **/
+    public void upsertUrl(String codenvyOnpremUrl) {
```

Comment:
I mean upsert, that is:
- insert if url doesn't exist
- update if url does exist